### DISCUSSION TITLE
Figgure out weater bot.getApi().getChatAdministrators(message->chat->id) is an administrator

### DISCUSSION THREAD

**Discussion by Beerlesklopfer** [reo7sp/tgbot-cpp#194] [Q&A] [answered]
Hello Community

to me there are two missing samples:

ONE:
 InlineKeyboardButton::Ptr btn_joinGroup(new InlineKeyboardButton);
  btn_joinGroup->text = "BlaBla";
  btn_joinGroup->url = "https://t.me/BlaBlaBot?startgroup=settings";
  btn_joinGroup->callbackData = "joinGroup";

This the one to join a grouo

TWO
How to figgure out wether the typing user is and admin an its ChatMember attributes

Any hints would be apreciated

**Comment by reo7sp** [comment, marked as answer]
`getChatAdministrators(chatId)` already returns only the owner and administrators. Compare each returned member user ID with the sender:

```cpp
bool senderIsAdmin = false;
for (const auto& member : bot.getApi().getChatAdministrators(message->chat->id)) {
    std::visit(
        [&](const auto& admin) {
            if (admin && admin->user && message->from && admin->user->id == message->from->id) {
                senderIsAdmin = true;
            }
        },
        member->value);
}
```

The bot must be allowed to obtain the administrator list. If you need particular permissions, inspect the `ChatMemberAdministrator` alternative; the owner is represented by `ChatMemberOwner` and has all administrator privileges.

The URL `https://t.me/BotName?startgroup=settings` only opens the add-to-group flow. A URL button does not send `callbackData`; `url` and `callbackData` are mutually exclusive button actions.
